tron/core/actionrun.py

```diff
@@ -1320,32 +1379,35 @@ def recover(self) -> Optional[KubernetesTask]:
-            secret_volumes=last_attempt.command_config.secret_volumes,
-            projected_sa_volumes=last_attempt.command_config.projected_sa_volumes,
-            volumes=last_attempt.command_config.extra_volumes,
-            cap_add=last_attempt.command_config.cap_add,
-            cap_drop=last_attempt.command_config.cap_drop,
-            task_id=last_attempt.kubernetes_task_id,
-            node_selectors=last_attempt.command_config.node_selectors,
-            node_affinities=last_attempt.command_config.node_affinities,
-            topology_spread_constraints=last_attempt.command_config.topology_spread_constraints,
-            pod_labels=build_labels(run_id=self.id, original_labels=last_attempt.command_config.labels),
-            pod_annotations=last_attempt.command_config.annotations,
-            service_account_name=last_attempt.command_config.service_account_name,
-            ports=last_attempt.command_config.ports,
-        )
+        # try/except block here is necessary cause if this fails, jobs will get resetted to 0 and we dont want that to happen
+        try:
+            task = k8s_cluster.create_task(
+                action_run_id=self.id,
+                command=last_attempt.rendered_command,
+                cpus=last_attempt.command_config.cpus,
+                mem=last_attempt.command_config.mem,
+                disk=last_attempt.command_config.disk,
+                docker_image=last_attempt.command_config.docker_image,
+                env=build_environment(original_env=last_attempt.command_config.env, run_id=self.id),
+                secret_env=last_attempt.command_config.secret_env,
+                field_selector_env=last_attempt.command_config.field_selector_env,
+                serializer=filehandler.OutputStreamSerializer(self.output_path),
+                secret_volumes=last_attempt.command_config.secret_volumes,
+                projected_sa_volumes=last_attempt.command_config.projected_sa_volumes,
+                volumes=last_attempt.command_config.extra_volumes,
+                cap_add=last_attempt.command_config.cap_add,
+                cap_drop=last_attempt.command_config.cap_drop,
+                task_id=last_attempt.kubernetes_task_id,
+                node_selectors=last_attempt.command_config.node_selectors,
+                node_affinities=last_attempt.command_config.node_affinities,
+                topology_spread_constraints=last_attempt.command_config.topology_spread_constraints,
+                pod_labels=build_labels(run_id=self.id, original_labels=last_attempt.command_config.labels),
+                pod_annotations=last_attempt.command_config.annotations,
+                service_account_name=last_attempt.command_config.service_account_name,
+                ports=last_attempt.command_config.ports,
+            )
+        except Exception:
+            log.exception(f"Unable to create task for ActionRun {self.id}")
```

Comment:
coming from https://github.com/Yelp/Tron/pull/1010#discussion_r1878875765

i kinda wonder if we should always have the "no retry on unknown" toggle enabled - this is now consistent with what the pre-existing code was doing, but present-luisp is not 100% sure that the pre-existing code is/was doing the safe/right thing :p

(i.e., i kinda wonder if we should crash tron if we get into this state)

Reply:
(that said, i think this should be fine for now?)

Reply:
yea, currently it will catch the exception log it instead of silently failing and resetting jobs as I was initially seeing

Reply:
what is the "no retry on unknown" toggle?

Reply:
sorry, that's just what i've been calling https://github.com/Yelp/Tron/pull/1000 in my head - it was something we added during the cluster migration to ensure that if something funky happened with the state during the move, we wouldn't accidentally run parallel runs of a job should tron think that the job was lost :p

Reply:
imo, maybe we do want to re-raise the exception here since failing to recover (what this exception handler catches) isn't the same as trying to recover a run that we no longer have k8s data for (what the handling for a null `task` is doing below) - it'd probably be easier to recover from a bad code push that caused errors here if tron didn't start rather than having tron transition the impacted runs to `UNKNOWN`

Reply:
sounds good, will reraise the exception